### PR TITLE
armagetronad: fix desktop icon

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #111467

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by numinit — CHANGES_REQUESTED** [review]
This breaks armagetronad-dedicated.

```
error: builder for '/nix/store/6h574vqslids9hy7rwf943215vm5n5yi-armagetronad-dedicated-0.2.9.1.1.drv' failed with exit code 1;
       last 10 log lines:
       >  /nix/store/5idwbbv23b6vnqdicx97s3hsgrwwnj7j-coreutils-9.4/bin/mkdir -p '/nix/store/9jsvgahvy7m99rsg06zd9h9jggib9vdm-armagetronad-dedicated-0.2.9.1.1/share/games/armagetronad-dedicated/language'
       >  /nix/store/5idwbbv23b6vnqdicx97s3hsgrwwnj7j-coreutils-9.4/bin/install -c -m 644  language/languages.txt language/english_base.txt language/english_base_notranslate.txt language/american.txt language/british.txt language/deutsch.txt language/french.txt language/spanish.txt language/polish_transliterated.txt language/polish.txt '/nix/store/9jsvgahvy7m99rsg06zd9h9jggib9vdm-armagetronad-dedicated-0.2.9.1.1/share/games/armagetronad-dedicated/language'
       > make  install-data-hook
       > make[3]: Entering directory '/build/source'
       > make[3]: Nothing to be done for 'install-data-hook'.
       > make[3]: Leaving directory '/build/source'
       > make[2]: Leaving directory '/build/source'
       > make[1]: Leaving directory '/build/source'
       > cp: cannot stat '/nix/store/9jsvgahvy7m99rsg06zd9h9jggib9vdm-armagetronad-dedicated-0.2.9.1.1/share/games/armagetronad/desktop/armagetronad.desktop': No such file or directory
       > /nix/store/bknngadwym46j65qs14ic2w79rpav888-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/6h574vqslids9hy7rwf943215vm5n5yi-armagetronad-dedicated-0.2.9.1.1.drv'.
error: 1 dependencies of derivation '/nix/store/mgdc9hi3w7hyj6akb0frkmm7sq8mlw7r-review-shell.drv' failed to build
```

File: pkgs/games/armagetronad/default.nix [hunk 1]
```diff
@@ -129,6 +129,12 @@ let
       nativeBuildInputs = [ autoconf automake gnum4 pkg-config which python3 ]
         ++ extraNativeBuildInputs;
 
+      postInstall = ''
```
Comment (numinit) [inline pkgs/games/armagetronad/default.nix:132]: Suggest: 

`lib.optionalString dedicatedServer ''`

File: pkgs/games/armagetronad/default.nix [hunk 2]
```diff
@@ -129,6 +129,12 @@ let
...
+        mkdir -p $out/share/{applications,icons/hicolor}
+        cp $out/share/games/armagetronad/desktop/armagetronad.desktop $out/share/applications/
```
Comment (numinit) [inline pkgs/games/armagetronad/default.nix:134]: Maybe a better idea to `ln -s` this and the below.

**Review by numinit — CHANGES_REQUESTED** [review]

File: pkgs/games/armagetronad/default.nix [hunk 3]
```diff
@@ -129,6 +129,12 @@ let
       nativeBuildInputs = [ autoconf automake gnum4 pkg-config which python3 ]
         ++ extraNativeBuildInputs;
 
+      postInstall = lib.optionalString (!dedicatedServer) ''
+        mkdir -p $out/share/{applications,icons/hicolor}
+        ln -s $out/share/games/armagetronad/desktop/armagetronad.desktop $out/share/applications/
```
Comment (numinit) [inline pkgs/games/armagetronad/default.nix:134]: For compat with 0.4:

`$out/share/games/armagetronad/desktop/armagetronad*.desktop`

**Review by numinit — CHANGES_REQUESTED** [review]

**Comment by numinit** [comment]
Seems a fairly obvious backport candidate. Thanks for the fix!

File: pkgs/games/armagetronad/default.nix (hunk 3 above)
Comment (numinit) [reply pkgs/games/armagetronad/default.nix:134]: `nix build '.#armagetronad-dedicated."0.4"'` btw.

**Review by fpletz — APPROVED** [review]

**Pull request merged**